fix: Support building in windows

Jordan-Hall opened the pull request:
I've not sorted the actual build release because im unsure about the CI pipeline. But you should be able to check this out and run on windows.


#5 tested on windows - should be close for linux too

talisk commented:
Thanks for your contribution. Very productive working! I will check it on Windows ASAP.

Jordan-Hall commented:
> Thanks for your contribution. Very productive working! I will check it on Windows ASAP.

Thank you. I've had to change it to work for both compared to the working version the night before. So something could of slipped through due to caching

Jordan-Hall commented:
I've got my linux laptop tomorrow. I'll check it out but i think its no longer macos and windows. Looking at what i had to change, it should work on linux. Hence i've added linux build script

talisk reviewed (commented):
Could you please write the building guide for windows in README?

talisk commented:
In your MR, can we get rid of WSL and use PowerShell with MinGW directly to build?

Jordan-Hall commented:
> Could you please write the building guide for windows in README?

Sure i'll add that later today. 

I assume you  are referring to use of rm to delete? Sure i'll change these to native windows command. When i install git I override the command with unix. I dont actually use WSL on my machine

neocarles commented:
Dear @Jordan-Hall how to fix this, i'm trying your's on windows and getting error

The system cannot find the path specified.
/d/lynx-devtool/packages/devtools-frontend-lynx
cp: cannot stat 'out/Default/gen/front_end/inspector.html': No such file or directory
sed: can't read output/inspector.html: No such file or directory
Error: Required files not found for packaging
 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed with exit code 1.

Jordan-Hall commented:
> Dear @Jordan-Hall how to fix this, i'm trying your's on windows and getting error
> 
> The system cannot find the path specified. /d/lynx-devtool/packages/devtools-frontend-lynx cp: cannot stat 'out/Default/gen/front_end/inspector.html': No such file or directory sed: can't read output/inspector.html: No such file or directory Error: Required files not found for packaging  ELIFECYCLE  Command failed with exit code 1.  ELIFECYCLE  Command failed with exit code 1.

Please accept my apologies. Needed to run pnpm install before the build. I've changed the setup.bat file. I've also changed the make file to do the same thing as it had the same issue (I copied order from that as i forgot exactly the order i got it to build in)

Jordan-Hall commented:
![{BFE2FF76-6981-41FF-8095-992F06597ADC}](https://github.com/user-attachments/assets/47de1bd6-719c-4b15-8732-feca2410266c)
@talisk  sorry for the delay and lots of changes :/


First .sh files wont run on windows so made them in JS so it's cross platform. 

Next gclient wouldn't run directly. Could used setx to add to path but thats then two workflows to run. Created a run gn to run the gclient from the path. Only way to work in windows.

talisk commented on packages/devtools-frontend-lynx/package.json:
```diff
@@ -19,30 +19,30 @@
     "generate-dark-mode-styles": "third_party/node/node.py --output scripts/dark_mode/generate_dark_theme_sheet.js",
-    "generate-protocol-resources": "scripts/deps/generate_protocol_resources.py && git cl format --js",
-    "install-deps": "scripts/deps/manage_node_deps.py",
+    "generate-protocol-resources": "python ./scripts/deps/generate_protocol_resources.py && git cl format --js",
+    "install-deps": "python ./scripts/deps/manage_node_deps.py",
```
There's a shebang in the top of these python file.

`#!/usr/bin/env vpython`

Therefore, we should not specify python commands explicitly.

https://chromium.googlesource.com/chromium/tools/depot_tools/master/vpython3

Maybe we should make sure that we can use vpython.

Jordan-Hall replied:
On windows using pnpm to run python script you cant just use the .py file. It requires to know what to use. I could do same as gclient and wrap it in JS to run. Alternative have a windows command. I cant see what actually calls these though on a search of the repo

talisk commented:
@Jordan-Hall 
Hi~

I have pushed a branch named `develop/windows-support` which from the main branch.
Could you please make this PR merges to this branch? 

And there're some additional work will be done, such as all build scripts will be in Python 3. We will continue our working on the new branch.

Jordan-Hall commented:
> erges to this branch

Sure thing That makes more sense to do in python. I've been away for python too long. Thank you for working on windows support

bgorati commented:
> Thanks for your contribution. Very productive working! I will check it on Windows ASAP.

**Windows**: Not working for me

C:\lynx-devtool\packages\lynx-devtool-cli:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @lynx-js/lynx-devtool-cli@0.0.1-beta.7 build: `tsc && npm run unzipResources`
Exit status 1
 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed with exit code 1.

talisk commented:
@Jordan-Hall Could you please fix the last 4 conflicts? And then I will merge to `develop/windows-support` branch.
